### PR TITLE
Add basic gc summarizer

### LINKED ISSUE #150: Add BasicGarbageCollectionSummarizer to JFRD
## Is your feature request related to a problem? Please describe.
Want access to detailed GC metrics for resource utilization analysis.

## Feature Description
Want access to these metrics:
- GC Major Collection Count
- GC Minor Collection Count
- GC Major Collection Time
- GC Minor Collection Time
- ~~GC ParNew Time~~.  Minor Collection Time is the same as ParNew. 

A lot of these details came out of [this slack conversation](https://newrelic.slack.com/archives/CV2EXQQA2/p1614867279019000) with Ben 

JFR produces the `jdk.GarbageCollection` GC event 

```
jdk.GarbageCollection {
  startTime = 11:52:18.076
  duration = 0.502 ms
  gcId = 859
  name = "G1New"
  cause = "G1 Evacuation Pause"
  sumOfPauses = 0.502 ms
  longestPause = 0.502 ms
}
```

To capture these vents, create a new `BasicGarbageCollectionSummarizer` in JFRD to accept these events and summarize into Summary metrics.    The `EVENT_NAME` of the summarizer will be `jdk.GarbageCollection`

There will be 4 new metrics created from this summarizer. 

`jfr.GarbageCollection.minorCount`
`jfr.GarbageCollection.minorDuration`
`jfr.GarbageCollection.majorCount`
`jfr.GarbageCollection.majorDuration`

The `name` field of the event specifies which Garbage Collector this data is for (i.e. G1, CMS, Parallel) and if it is Major (old) or Minor(young) event.    "G1New" here means a young event. When multiple garbage collectors are in use, for example when Parallel (`name` = ParNew) and G1 (`name` = G1New), both events will feed into one metric - `jfr.GarbageCollection.minorCount`

 The `name` field is important as it will determine if JFRD supports including the event in the summary metric. 

First

 Test JFR with all 3 mainstream collectors  - Parallel, CMS, G1 - to generate and examine the raw JFR events. Need to see the structure of the data, especially the `name` value.  Verify the name for Old and Young events for all three collectors. 

Second 

Add the new summarizer to JFRD and manually test for all three different collector scenarios. 

## Priority
Must Have

### PR DESCRIPTION
**Pull request by jasonjkeller**
Adds `BasicGarbageCollectionSummarizer` to create `jfr.GarbageCollection.minorDuration` and `jfr.GarbageCollection.majorDuration` metrics. These metrics include the count.

```
        {
          "app.name": "jfr-daemon",
          "collector.name": "JFR-Uploader",
          "endTimestamp": 1615421055034,
          "entity.guid": "MjIxMjg2NHxFWFR8U0VSVklDRXwtODc2Mzc1ODMwMzk3NDAwNDM2OQ",
          "entity.name": "jfr-daemon",
          "host.hostname": "C02XQ1RKJG5J/192.168.1.23",
          "instrumentation.name": "JFR",
          "instrumentation.provider": "JFR-Uploader",
          "jfr.GarbageCollection.minorDuration": {
            "type": "summary",
            "count": 490,
            "sum": 1202,
            "min": 0,
            "max": 11
          },
          "metricName": "jfr.GarbageCollection.minorDuration",
          "newrelic.source": "metricAPI",
          "nr.entity.type": "SERVICE",
          "service.name": "jfr-daemon",
          "timestamp": 1615421045052
        }
```

```
        {
          "app.name": "jfr-daemon",
          "collector.name": "JFR-Uploader",
          "endTimestamp": 1615421054939,
          "entity.guid": "MjIxMjg2NHxFWFR8U0VSVklDRXwtODc2Mzc1ODMwMzk3NDAwNDM2OQ",
          "entity.name": "jfr-daemon",
          "host.hostname": "C02XQ1RKJG5J/192.168.1.23",
          "instrumentation.name": "JFR",
          "instrumentation.provider": "JFR-Uploader",
          "jfr.GarbageCollection.majorDuration": {
            "type": "summary",
            "count": 501,
            "sum": 20280,
            "min": 38,
            "max": 50
          },
          "metricName": "jfr.GarbageCollection.majorDuration",
          "newrelic.source": "metricAPI",
          "nr.entity.type": "SERVICE",
          "service.name": "jfr-daemon",
          "timestamp": 1615421045052
        }
```

Resolves #150

### CODE REVIEW

File: jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        minorGcEndTimeMs = ev.getStartTime().toEpochMilli();
+        minorGcDurationSummarizer.accept(ev);
+        minorGcCount++;
+      } else if (MAJOR_GC_NAMES.contains(name)) {
+        majorGcEndTimeMs = ev.getStartTime().toEpochMilli();
+        majorGcDurationSummarizer.accept(ev);
+        majorGcCount++;
+      } else
+        // Ignore events with GC name: GCNameEndSentinel, N/A, Shenandoah, Z or anything unexpected
+        logger.warn("Ignoring unsupported " + EVENT_NAME + " event: " + name);
```
Comment (jasonjkeller) [inline jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java:101]: @kittylyst You mentioned wanting to log when we skip an event with an unsupported GC name, so I added this. However, there was no prior precedent of doing any logging from the jfr-mappers module which means that the slf4j dependency had to be added. Is this definitely something that we want to do? If not, just let me know and I can roll back the logging changes.

File: jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        new Summary(
+            "jfr.GarbageCollection.majorDuration",
+            majorGcCount,
+            majorGcDurationSummarizer.getDurationMillis(),
+            majorGcDurationSummarizer.getMinDurationMillis(),
+            majorGcDurationSummarizer.getMaxDurationMillis(),
+            startTimeMs,
+            majorGcEndTimeMs,
+            attr);
+    return Stream.of(minorGcDuration, majorGcDuration);
```
Comment (jasonjkeller) [inline jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java:129]: @kittylyst Is this the correct approach, always creating both metrics? Or should there be logic that only returns a stream of summarizers that have a count > 0? 

On the one hand, this approach means that both metrics will always have _some_ value and nice continuous lines in the charts. On the other, we would potentially be sending up metrics with the default value of 0 for count/sum/min/max.

I'd love to get some feedback on which approach is more desirable.

Comment (XiXiaPdx) [reply jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java:129]: @kittylyst  IMO, from a useability perspective, `0` and a flat line is information that we are monitoring major GC events and that event hasn't happened yet.  That is reassuring and removes doubt "Is jfr recording major gc events??"    2️⃣ 💴

Comment (kittylyst) [reply jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java:129]: 👍 to @XiXiaPdx comments

File: jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java (hunk 1 above)
Comment (kittylyst) [reply jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java:101]: I had wanted to keep the transformation library free from logging. However, we are still tightly coupled to the telemetry library, so I'm not sure that matters that much right now. We can revisit later - for now, let's add the dependency

**Review by kittylyst — APPROVED** [review]
LGTM.

**Pull request merged**